docs/how_to_add_test_for_rector_rule.md

```diff
@@ -0,0 +1,119 @@
+# How to Add Test for Rector Rule
+
+## 1. Detect the Rector Rule
+
+Run Rector only on 1 directory, or better 1 file.
+
+```bash
+vendor/bin/rector process compiler/src/PhpScoper/StaticEasyPrefixer.php
+```
+
+See "Applied rules" under the diff:
+
+![Applied Rules](/docs/images/docs_applied_rules.png)
+
+Our rule in this case is: `Rector\SOLID\Rector\Class_\FinalizeClassesWithoutChildrenRector`
+
+## 2. Detect the Minimal File
+
+Usually the Rector diff output is long and contains many other errors related to other rules. It's a mess, we can't use that for test fixture. We need to find **1 responsible line**.
+
+The best way is to copy the file to local code, e.g. `app/SomeFile.php` a put only the broken line there.
+
+In our case, all we need is:
+
+```php
+class StaticEasyPrefixer
+{
+}
+```
+
+Then run Rector again to confirm:
+
+```bash
+vendor/bin/rector process app/SomeFile.php
+```
+
+Do we have the same diff? Great!
+
+## 3. Find the Rector Test Case
+
+Now we need to find the test case. The test case name is rule + `Test` suffix.
+
+`FinalizeClassesWithoutChildrenRector`
+
+↓
+
+`FinalizeClassesWithoutChildrenRectorTest` (test class)
+
+↓
+
+`FinalizeClassesWithoutChildrenRectorTest.php` (test file)
+
+Right here:
+
+![Rule Test Case](/docs/images/docs_rule_test_case.png)
+
+## 4. Add Change or No-Change fixture
+
+Next to test case, there is `/Fixture` directory, there we create our test fixture file, e.g. `add_final.php.inc`. The `.php.inc` is there on purpose, so the file is hidden from coding standard tools and static analysis.
```

Comment:
You may want to say that a test can have many fixture tests, not just one.

Btw, I find sentence `"there we create our test fixture file"` a bit confusing myself: what does "fixture" mean? Is it a word used everywhere concerning testing, or just for Rector? Is it the same as "test cases"?

I've seen that many tests are called `fixture.php.inc`, is that supposed to be the main one? Or there's no need even to call it `fixture.php.inc` (say, when there's only 1 test)?

Reply:
Thanks for feedback, I'll send updated version to address your comments.